tests/e2e/specs/shopper/filter-products-by-rating.test.ts

```diff
@@ -204,4 +207,120 @@ describe.skip( `${ block.name } Block`, () => {
 			);
 		} );
 	} );
+
+	describe( 'with Product Query Block', () => {
+		let editorPageUrl = '';
+		let frontedPageUrl = '';
+
+		useTheme( 'emptytheme' );
+		beforeAll( async () => {
+			await switchUserToAdmin();
+			await createNewPost( {
+				postType: 'post',
+				title: block.name,
+			} );
+
+			await insertBlock( 'Product Query' );
+			await insertBlock( block.name );
+			await page.waitForNetworkIdle();
+			await publishPost();
+
+			editorPageUrl = page.url();
+			frontedPageUrl = await page.evaluate( () =>
+				wp.data.select( 'core/editor' ).getPermalink()
+			);
+			await page.goto( frontedPageUrl );
+		} );
+
+		it( 'should render', async () => {
+			await expect( page ).toMatchElement( block.class );
+		} );
+
+		it( 'should render products', async () => {
+			const products = await page.$$(
+				selectors.frontend.queryProductsList
+			);
+
+			expect( products ).toHaveLength( 5 );
+		} );
+
+		it( 'should show only products that match the filter', async () => {
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+
+			await page.waitForSelector( block.class + '.is-loading', {
+				hidden: true,
+			} );
+
+			expect( isRefreshed ).not.toBeCalled();
+
+			await page.waitForSelector( selectors.frontend.fiveStarInput );
+
+			await Promise.all( [
+				page.waitForNavigation(),
+				page.click( selectors.frontend.fiveStarInput ),
+			] );
+
+			const products = await page.$$(
+				selectors.frontend.queryProductsList
+			);
+			const pageURL = page.url();
+			const parsedURL = new URL( pageURL );
+
+			expect( isRefreshed ).toBeCalledTimes( 1 );
+			expect( products ).toHaveLength( FIVE_STAR_PRODUCTS_AMOUNT );
+			expect( parsedURL.search ).toEqual(
+				block.urlSearchParamWhenFilterIsApplied
+			);
+		} );
+
+		it( 'should refresh the page only if the user click on button', async () => {
+			await page.goto( editorPageUrl );
+			await openBlockEditorSettings();
+			await selectBlockByName( block.slug );
+			await openBlockEditorSettings();
+			await page.waitForXPath(
+				block.selectors.editor.filterButtonToggle
+			);
+
+			const [ filterButtonToggle ] = await page.$x(
+				selectors.editor.filterButtonToggle
+			);
```

Comment:
`page.waitForXPath()` returns an element, so we can combine these calls into one, something like:

```suggestion
			const filterButtonToggle = await page.waitForXPath(
				block.selectors.editor.filterButtonToggle
			);
```

By the way, we introduced some new utilities to interact with inspector settings, can you take a look at [them](https://github.com/woocommerce/woocommerce-blocks/blob/8c857e2422d68479e7f5d798184fa19931e1eb07/tests/e2e/specs/backend/product-query/advanced-filters.ts#L149-L155) to see if we can use them here?

https://github.com/woocommerce/woocommerce-blocks/blob/8c857e2422d68479e7f5d798184fa19931e1eb07/tests/e2e/specs/backend/product-query/advanced-filters.ts#L149-L155